src/bt_peer_connection.cpp

```diff
@@ -1906,7 +1906,7 @@ namespace libtorrent
 		// if we're finished and this peer is uploading only
 		// disconnect it
 		if (t->is_finished() && upload_only()
-			&& m_settings.get_bool(settings_pack::close_redundant_connections)
```

Comment:
don't you need to check both of these? or do you update the torrent settings if the global change?